accounts-db/src/accounts_index/in_mem_accounts_index.rs

```diff
@@ -884,9 +890,25 @@ impl<T: IndexValue, U: DiskIndexValue + From<T> + Into<T>> InMemAccountsIndex<T,
         startup: bool,
         current_age: Age,
         ages_flushing_now: Age,
+        max_evictions: Option<usize>,
     ) -> Vec<(Pubkey, /*is_dirty*/ bool)> {
         let mut possible_evictions = Vec::new();
         for (k, v) in iter {
+            // Check if we've reached the max evictions limit
+            if let Some(max) = max_evictions {
+                if possible_evictions.len() >= max {
+                    break;
+                }
+            }
+
+            // Age wrapping consideration: Age is u8 (0-255) and increments every AGE_MS (2 seconds).
+            // After 256 age increments (512 seconds), age wraps around to 0, which could cause
+            // wrapping_sub to incorrectly calculate the age difference for very old entries that
+            // fall outside the eviction window (current_age - ages_flushing_now).
+            // This means entries that remain unused for 512+ seconds may not be considered
+            // candidates for eviction. However, this is acceptable because:
+            // 1. The entry count limit is sized large enough to hold all frequently-accessed accounts
+            // 2. Other cold accounts within the eviction window will still be evicted normally
```

Comment:
I think the ages wrap around is annoying, as specific old ages will wrap and appear "new" and thus not evicted. I agree with this PR it's fine. Long term I think we should remove/replace the ages concept.